File: app/src/main/java/com/zayed/admin/yummyrecipes/MuttonGridView.java
package com.zayed.admin.yummyrecipes;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.GridView;

public class MuttonGridView extends Activity {

    private int [] MuttonGridImage = {R.drawable.muttoncutleta,R.drawable.muttonseekh,R.drawable.muttonkarahi,R.drawable.muttonkeema,R.drawable.muttonkorma,R.drawable.roganjosh,R.drawable.muttonbiryania,R.drawable.muttonpuloa};
    private String [] MuttonGridNames = {"Cutlet Kebab","Seekh Kebab","Mutton Karahi","Mutton Keema","Mutton Korma","Rogan Josh","Mutton Biryani","Mutton Pulao"};
    private GridView gridView;
    private MuttonGridAdapter muttonGridAdapter;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.layout_mutton);

        gridView = (GridView)findViewById(R.id.muttongrid);
        muttonGridAdapter = new MuttonGridAdapter(this,MuttonGridImage,MuttonGridNames);
        gridView.setAdapter(muttonGridAdapter);


        gridView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {

                switch (position)
                {
                    case 0:
                        Intent intent = new Intent(MuttonGridView.this,CutletKebab.class);
                        startActivity(intent);
                        break;

                    case 1:
                        Intent intent1 = new Intent(MuttonGridView.this,SeekhKebabrecipes.class);
                        startActivity(intent1);
                        break;

                    case 2:
                        Intent intent2 = new Intent(MuttonGridView.this,MuttonkarahiRecipes.class);
                        startActivity(intent2);
                        break;

                    case 3:
                        Intent intent3 = new Intent(MuttonGridView.this,MuttonKeemaRecipes.class);
                        startActivity(intent3);
                        break;

                    case 4:

                    Intent intent4 = new Intent(MuttonGridView.this,MuttonKorma.class);
                    startActivity(intent4);
                    break;

                    case 5:
                        Intent intent5 = new Intent(MuttonGridView.this,RoganJoshRecipes.class);
                        startActivity(intent5);
                        break;

                    case 6:
                        Intent intent6 = new Intent(MuttonGridView.this,MuttonBiryani.class);
                        startActivity(intent6);
                        break;

                    case 7:
                        Intent intent7 = new Intent(MuttonGridView.this,MuttonPuloaRecipes.class);
                        startActivity(intent7);
                        break;
                }

            }
        });
    }
}

File: app/src/main/java/com/zayed/admin/yummyrecipes/MuttonGridView.kt
package com.zayed.admin.yummyrecipes

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.AdapterView.OnItemClickListener
import android.widget.GridView
import com.zayed.admin.yummyrecipes.MuttonGridView

class MuttonGridView : Activity() {
    private val MuttonGridImage = intArrayOf(R.drawable.muttoncutleta, R.drawable.muttonseekh, R.drawable.muttonkarahi, R.drawable.muttonkeema, R.drawable.muttonkorma, R.drawable.roganjosh, R.drawable.muttonbiryania, R.drawable.muttonpuloa)
    private val MuttonGridNames = arrayOf("Cutlet Kebab", "Seekh Kebab", "Mutton Karahi", "Mutton Keema", "Mutton Korma", "Rogan Josh", "Mutton Biryani", "Mutton Pulao")
    private var gridView: GridView? = null
    private var muttonGridAdapter: MuttonGridAdapter? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.layout_mutton)
        gridView = findViewById<View>(R.id.muttongrid) as GridView
        muttonGridAdapter = MuttonGridAdapter(this, MuttonGridImage, MuttonGridNames)
        gridView!!.adapter = muttonGridAdapter
        gridView!!.onItemClickListener = OnItemClickListener { parent, view, position, id ->
            when (position) {
                0 -> {
                    val intent = Intent(this@MuttonGridView, CutletKebab::class.java)
                    startActivity(intent)
                }
                1 -> {
                    val intent1 = Intent(this@MuttonGridView, SeekhKebabrecipes::class.java)
                    startActivity(intent1)
                }
                2 -> {
                    val intent2 = Intent(this@MuttonGridView, MuttonkarahiRecipes::class.java)
                    startActivity(intent2)
                }
                3 -> {
                    val intent3 = Intent(this@MuttonGridView, MuttonKeemaRecipes::class.java)
                    startActivity(intent3)
                }
                4 -> {
                    val intent4 = Intent(this@MuttonGridView, MuttonKorma::class.java)
                    startActivity(intent4)
                }
                5 -> {
                    val intent5 = Intent(this@MuttonGridView, RoganJoshRecipes::class.java)
                    startActivity(intent5)
                }
                6 -> {
                    val intent6 = Intent(this@MuttonGridView, MuttonBiryani::class.java)
                    startActivity(intent6)
                }
                7 -> {
                    val intent7 = Intent(this@MuttonGridView, MuttonPuloaRecipes::class.java)
                    startActivity(intent7)
                }
            }
        }
    }
}